snapcraft/projects.py

```diff
@@ -64,6 +65,114 @@ def _validate_command_chain(command_chains: Optional[List[str]]) -> Optional[Lis
     return command_chains
 
 
+def _validate_architectures(architectures):
+    """Expand and validate architecture data.
+
+    Validation includes:
+        - The list cannot be a combination of strings and Architecture objects.
+        - The same architecture cannot be defined in multiple `build-on` or
```

Comment:
Whoops.  The same architecture can be used in multiple `build-on` fields.  Will fix tomorrow.